src/main/drivers/video_max7456.c

```diff
@@ -0,0 +1,620 @@
+    // (See DEBUG_MAX7456_DM_UPDATE)
+
+    max7456State.vSyncDetected = false;
+
+    for (int y = 0; y < textScreen->height; y++) {
+        unsigned int rowOffset = (y * textScreen->width);
+        char *buffer = &screenBuffer[rowOffset];
+
+        if (y == 8) {
+            MAX7456_TIME_SECTION_END(1);
+            // wait for next frame to send the remaining rows
+            max7456State.vSyncDetected = false;
+        }
+
+        while (!max7456State.vSyncDetected) {
+            // Wait for VSYNC pulse and ISR to update the state.
+        }
+
+        if (y == 0) {
+            MAX7456_TIME_SECTION_BEGIN(1);
+        } else if (y == 8) {
+            MAX7456_TIME_SECTION_BEGIN(2);
+        }
+
+        for (int x = 0; x < textScreen->width; x++) {
+            // TODO ensure 0xFF is replaced with ' ' to avoid early termination of auto-increment mode.
+            spiTransferByte(MAX7456_SPI_INSTANCE, MAX7456_REG_DMDI);
+            spiTransferByte(MAX7456_SPI_INSTANCE, *buffer++);
+        }
+    }
+
+    spiTransferByte(MAX7456_SPI_INSTANCE, MAX7456_REG_DMDI);
+    spiTransferByte(MAX7456_SPI_INSTANCE, 0xFF); // terminate auto-increment
+
+    spiTransferByte(MAX7456_SPI_INSTANCE, MAX7456_REG_DMM);
+    spiTransferByte(MAX7456_SPI_INSTANCE, 0x00); // disable 8bit mode and auto increment, enabled above.
+
+    MAX7456_TIME_SECTION_END(2);
+
+    DISABLE_MAX7456;
+
+
+}
+
+// show the entire font in the middle of the screen.
+
+
+void max7456_setCharacterAtPosition(uint8_t x, uint8_t y, uint8_t c)
+{
+    uint32_t linepos;
+    uint8_t char_address_hi, char_address_lo;
+
+    //find start address position
+    linepos = y * 30 + x;
+
+    // divide 16 bits into hi & lo uint8_t
+    char_address_hi = linepos >> 8;
+    char_address_lo = linepos;
+
+    ENABLE_MAX7456;
+
+    spiTransferByte(MAX7456_SPI_INSTANCE, MAX7456_REG_DMAH); // set start address high
+    spiTransferByte(MAX7456_SPI_INSTANCE, char_address_hi);
+
+    spiTransferByte(MAX7456_SPI_INSTANCE, MAX7456_REG_DMAL); // set start address low
+    spiTransferByte(MAX7456_SPI_INSTANCE, char_address_lo);
+
+    spiTransferByte(MAX7456_SPI_INSTANCE, MAX7456_REG_DMDI);
+    spiTransferByte(MAX7456_SPI_INSTANCE, c);
+
+    DISABLE_MAX7456;
+}
+
+void max7456_showFont(void)
+{
+    int c = 0, y = 2;
+    while (c < 256) {
+        for (uint8_t x = 0; x < 24 && c < 256; x++) {
+            max7456_setCharacterAtPosition(x + 3, y, c++);
+        }
+        y++;
+    }
+}
+
+void max7456_clearScreen(void)
+{
+    uint8_t dmmStatus = max7456_read(MAX7456_REG_DMM_READ);
+
+    dmmStatus |= MAX7456_DMM_BIT_CLEAR;
+    max7456_write(MAX7456_REG_DMM, dmmStatus);
+    delay(20);
+}
+
+void max7456_clearScreenAtNextVSync(void)
+{
+    uint8_t dmmStatus = max7456_read(MAX7456_REG_DMM_READ);
+
+    dmmStatus |= MAX7456_DMM_BIT_CLEAR;
+    dmmStatus |= MAX7456_DMM_BIT_VSYNC_CLEAR;
+    max7456_write(MAX7456_REG_DMM, dmmStatus);
+    delay(20);
```

Comment:
delay does not make much sense here, clear will happen at next VSYNC (actually, VSYNC will happen within 20ms)